### ISSUE TITLE
Webview: FAILED waiting for webview ready event; closing webview...  (VsCode)

### ISSUE THREAD

**Issue by AchilleLacoin** [TeamCodeStream/CodeStream#12]
After signing-in, the view of messages is never shown.
I don't know where the log files are stored, so here is a screenshot of the issue :

![codestream](https://user-images.githubusercontent.com/36007425/47716590-79d93e00-dc43-11e8-8992-9b1648ea3885.png)

**Comment by craigharman** [comment]
I have the same issue.... log dump below:

[2018-10-31 04:37:23:691] CodeStream v0.20.0-21 in VS Code (v1.28.2) starting... 
[2018-10-31 04:37:23:839] CodeStream v0.20.0-21 started • 308 ms 
[2018-10-31 04:37:23:935] Signing <email-removed> into CodeStream (https://api.codestream.com) 
[2018-10-31 04:37:31:766] AgentConnection.sendRequest(codeStream/users/me/unreads) 
[2018-10-31 04:37:31:768] <email-removed> signed into CodeStream (https://api.codestream.com); userId=<removed>, teamId=<removed> 
[2018-10-31 04:37:31:769] AgentConnection.sendRequest(codeStream/textDocument/markers) 
[2018-10-31 04:37:31:770] AgentConnection.sendRequest(codeStream/repos) 
[2018-10-31 04:37:31:770] AgentConnection.sendRequest(codeStream/streams) 
[2018-10-31 04:37:31:770] AgentConnection.sendRequest(codeStream/teams) 
[2018-10-31 04:37:31:770] AgentConnection.sendRequest(codeStream/users/me/unreads) 
[2018-10-31 04:37:31:770] AgentConnection.sendRequest(codeStream/users) 
[2018-10-31 04:37:31:771] AgentConnection.sendRequest(codeStream/users/me/preferences) 
[2018-10-31 04:37:32:059] WebviewPanel: FAILED posting codestream:interaction:blur to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:32:490] AgentConnection.sendRequest(codeStream/users/me/preferences) undefined
Error: Request codeStream/users/me/preferences failed with message: An API error occurred: missing_scope
[2018-10-31 04:37:32:749] WebviewPanel: Received message codestream:request(1:AwFzrlFYTGi9CmIZJ6U-nQ):bootstrap from the webview 
[2018-10-31 04:37:32:749] WebviewPanel: Bootstrapping webview... SignedIn=true
[2018-10-31 04:37:32:750]  
Error: Request codeStream/users/me/preferences failed with message: An API error occurred: missing_scope
[2018-10-31 04:37:32:751] WebviewPanel: Completed posting codestream:response(1:AwFzrlFYTGi9CmIZJ6U-nQ) to the webview 
[2018-10-31 04:37:34:431] AgentConnection.onDataChanged(streams) 
[2018-10-31 04:37:34:685] WebviewPanel: Attempting to send codestream:data(streams) to the webview... 
[2018-10-31 04:37:34:685] WebviewPanel: FAILED posting codestream:data(streams) to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:36:645] AgentConnection.onDataChanged(streams) 
[2018-10-31 04:37:36:967] WebviewPanel: Attempting to send codestream:data(streams) to the webview... 
[2018-10-31 04:37:36:967] WebviewPanel: FAILED posting codestream:data(streams) to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:38:837] AgentConnection.onDataChanged(streams) 
[2018-10-31 04:37:39:156] WebviewPanel: Attempting to send codestream:data(streams) to the webview... 
[2018-10-31 04:37:39:156] WebviewPanel: FAILED posting codestream:data(streams) to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:40:872] WebviewPanel: FAILED posting codestream:interaction:focus to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:41:057] AgentConnection.onDataChanged(streams) 
[2018-10-31 04:37:41:337] WebviewPanel: Attempting to send codestream:data(streams) to the webview... 
[2018-10-31 04:37:41:337] WebviewPanel: FAILED posting codestream:data(streams) to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:43:270] AgentConnection.onDataChanged(streams) 
[2018-10-31 04:37:43:523] WebviewPanel: Attempting to send codestream:data(streams) to the webview... 
[2018-10-31 04:37:43:523] WebviewPanel: FAILED posting codestream:data(streams) to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:45:492] AgentConnection.onDataChanged(streams) 
[2018-10-31 04:37:45:747] WebviewPanel: Attempting to send codestream:data(streams) to the webview... 
[2018-10-31 04:37:45:747] WebviewPanel: FAILED posting codestream:data(streams) to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:47:738] AgentConnection.onDataChanged(streams) 
[2018-10-31 04:37:47:991] WebviewPanel: Attempting to send codestream:data(streams) to the webview... 
[2018-10-31 04:37:47:991] WebviewPanel: FAILED posting codestream:data(streams) to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:48:810] AgentConnection.onDataChanged(streams) 
[2018-10-31 04:37:49:064] WebviewPanel: Attempting to send codestream:data(streams) to the webview... 
[2018-10-31 04:37:49:064] WebviewPanel: FAILED posting codestream:data(streams) to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:37:54:793] WebviewPanel: FAILED posting codestream:interaction:blur to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:38:01:928] Webview: FAILED waiting for webview ready event; closing webview... 
[2018-10-31 04:38:01:928] Webview: CANCELLED waiting for webview ready event 
[2018-10-31 04:38:46:684] AgentConnection.sendRequest(codeStream/post/prepareWithCode) 
[2018-10-31 04:38:46:730] AgentConnection.sendRequest(codeStream/repos) 
[2018-10-31 04:38:46:731] AgentConnection.sendRequest(codeStream/streams) 
[2018-10-31 04:38:46:731] AgentConnection.sendRequest(codeStream/teams) 
[2018-10-31 04:38:46:731] AgentConnection.sendRequest(codeStream/users/me/unreads) 
[2018-10-31 04:38:46:731] AgentConnection.sendRequest(codeStream/users) 
[2018-10-31 04:38:46:731] AgentConnection.sendRequest(codeStream/users/me/preferences) 
[2018-10-31 04:38:47:224] AgentConnection.sendRequest(codeStream/users/me/preferences) undefined
Error: Request codeStream/users/me/preferences failed with message: An API error occurred: missing_scope
[2018-10-31 04:38:47:535] WebviewPanel: Received message codestream:request(1:5-rlPQE7QT6aRDmUBHM2sQ):bootstrap from the webview 
[2018-10-31 04:38:47:535] WebviewPanel: Bootstrapping webview... SignedIn=true
[2018-10-31 04:38:47:535]  
Error: Request codeStream/users/me/preferences failed with message: An API error occurred: missing_scope
[2018-10-31 04:38:47:536] WebviewPanel: Completed posting codestream:response(1:5-rlPQE7QT6aRDmUBHM2sQ) to the webview 
[2018-10-31 04:39:06:898] WebviewPanel: FAILED posting codestream:interaction:blur to the webview; Webview is invisible and can't receive messages 
[2018-10-31 04:39:16:735] Webview: FAILED waiting for webview ready event; closing webview... 
[2018-10-31 04:39:16:736] Webview: CANCELLED waiting for webview ready event 
[2018-10-31 04:39:16:736]

**Comment by eamodio** [comment]
@AchilleLacoin @craigharman Sorry for the issues you both are running into. While we aren't yet sure why you are running into scope issues (as we can't reproduce the issue ourselves), we will be publishing an update (likely tomorrow) that should address this issue and allow you to use CodeStream.

**Comment by planteater** [comment]
Fixed in version 0.21

**Comment by craigharman** [comment]
Thanks @planteater but I have updated the plugin and still get an error

(node:48328) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: An API error occurred: missing_scope
(node:48328) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[Warn  - 10:13:30 AM] [2018-11-02 02:13:30:714] SLACK: users.prefs.get: TIMEOUT 10s exceeded

**Comment by planteater** [comment]
@craigharman - Could I trouble you to try again and then send us the complete contents of the Output view? Thanks! We'll get to the bottom of this.

**Comment by planteater** [comment]
@craigharman @AchilleLacoin - v0.22 just hit the marketplace and should fix this issue. Let us know!

**Comment by craigharman** [comment]
Thanks for our help @planteater we are up and running now!

**Comment by DanSenn1** [comment]
I'm using v0.23.0 with vscode v1.28.2 on windows 10 and having a similar problem using the slack integration - logs are attached.
[codestream-log.txt](https://github.com/TeamCodeStream/CodeStream/files/2573030/codestream-log.txt)
[codestream-agent-log.txt](https://github.com/TeamCodeStream/CodeStream/files/2573031/codestream-agent-log.txt)

**Comment by eamodio** [comment]
@DanSenn1 Can you try signing out of CodeStream (using `CodeStream: Sign Out` from the command palette) and then re-auth with Slack? We believe there is a mismatch from when you generated the original token, to our requirements today.

Please let me know if this resolves the issue for you. Thanks!

**Comment by DanSenn1** [comment]
@eamodio Thanks for getting back to me.  I signed out and back in and was able to see the channel list (after about 30 minutes of loading), but I couldn't interact with it at all.  I tried to close and reopen the window and it timed out again trying to open.  I was also unable to use the "add CodeStream comment" function, presumably because the webview is timing out.  The log from after I came back and the channels were loaded is attached.
[codestream-after-reauth.txt](https://github.com/TeamCodeStream/CodeStream/files/2574345/codestream-after-reauth.txt)
